### PR TITLE
feat(RELEASE-2488): use python script for embargo-check task

### PR DESCRIPTION
**Pull request by querti**
Assisted-by: Cursor

## Describe your changes
Replace the inline bash check-issues and check-cves steps
with a single step running the embargo_check.py script from
release-service-utils. Consolidate integration tests into one
happy-path pipeline backed by mock HTTP server and mock
binaries, with edge cases now covered by unit tests in the
utils repo.
## Relevant Jira
https://redhat.atlassian.net/browse/RELEASE-2488
## Checklist before requesting a review
- [x] I have marked as draft or added `do not merge` label if there's a dependency PR
  - If you want reviews on your draft PR, you can add reviewers or add the `release-service-maintainers` handle if you are unsure who to tag
- [x] My commit message includes `Signed-off-by: My name <email>`
- [x] I read CONTRIBUTING.MD and [commit formatting](CONTRIBUTING.md#commit-message-formatting-and-standards)
- [x] I have run the README.md generator script in `.github/scripts/readme_generator.sh` and verified the results using `.github/scripts/check_readme.sh`
- [x] If an AI agent was used, I marked that via a commit footer like `Assisted-By: Cursor`

### CODE REVIEW

File: tasks/managed/embargo-check/tests/python_mocks/sitecustomize.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+"""Redirect Jira API requests to the local mock server for Tekton tests.
```
Comment (FilipNikolovski) [inline tasks/managed/embargo-check/tests/python_mocks/sitecustomize.py:1]: Why is the file named `sitecustomize.py`, i dont see any references to it in here?

**Review by FilipNikolovski — DISMISSED** [review]
One minor question, otherwise lgtm.

File: tasks/managed/embargo-check/tests/python_mocks/sitecustomize.py (hunk 1 above)
Comment (querti) [reply tasks/managed/embargo-check/tests/python_mocks/sitecustomize.py:1]: It was created by Claude and it seems to be a not very well known Python feature. Explanation from Claude:

sitecustomize is a [standard Python mechanism](https://docs.python.org/3/library/site.html) — the interpreter automatically imports any module with that name found on sys.path at startup, before user code runs. The test framework in .github/scripts/render_python_task_mocks_from_yaml.py picks up all .py files from tests/python_mocks/, drops them into a temp dir on PYTHONPATH, so naming it sitecustomize.py makes Python load the monkey-patch transparently — no explicit import needed.

I couldn't use the existing render_python_task_mocks_from_yaml.py because the code has an allowlist for the Jira hostnames and the hostname substitution would cause the task to fail.

Comment (johnbieren) [reply tasks/managed/embargo-check/tests/python_mocks/sitecustomize.py:1]: You can certainly update render_python_task_mocks_from_yaml if that makes more sense

**Review by johnbieren — DISMISSED** [review]

**Review by johnbieren — APPROVED** [review]

**Review by ach912 — APPROVED** [review]
lgtm

**Comment by ach912** [comment]
## Production Approval Record

| Field | Value |
|---|---|
| **Action** | APPROVED |
| **Reviewer** | @ach912 |
| **Timestamp** | 2026-08-04T15:56:57.191Z |
| **Criteria Override** | AI review findings not addressed |

Approved

**Review by elenagerman — APPROVED** [review]
